test/gtest/ucp/test_ucp_wakeup.cc

```diff
@@ -174,3 +175,266 @@ UCS_TEST_P(test_ucp_wakeup, signal)
+    for (; i != tl_names.end(); ++i) {
+        if (transports.end() !=
+            std::find_if(transports.begin(),
+                         transports.end(),
+                         std::bind2nd(std::equal_to<std::string>(), *i))) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void
+test_ucp_wakeup_events::do_tx_rx_events_test(const std::vector<std::string>& transports,
+                                             unsigned events)
+{
+    int efd[2];
+    struct pollfd polled[2];
+    ucp_test_base::entity *p_entity[2] = { &sender(), &receiver() };
+    const ucp_datatype_t DATATYPE = ucp_dt_make_contig(1);
+    const uint64_t TAG = 0xdeadbeef;
+    uint64_t send_data = 0x12121212;
+    const size_t msg_count = 1001;
+    std::vector<void *> req(2*msg_count, NULL);
+
+    /* UD based transports may cause extra events */
+    const char*  ud_tls[]   = { "\\ud", "\\ud_mlx5" };
+    const size_t ud_tls_cnt = (sizeof(ud_tls) / sizeof(ud_tls[0]));
+    const bool has_ud = is_any_tl_inuse(transports,
+                                        std::vector<std::string>(ud_tls,
+                                                                 ud_tls +
+                                                                 ud_tls_cnt));
+
+    polled[0].events = polled[1].events = POLLIN;
+
+    p_entity[0]->connect(p_entity[1], get_ep_params());
+    p_entity[1]->connect(p_entity[0], get_ep_params());
+
+    ASSERT_UCS_OK(ucp_worker_get_efd(p_entity[0]->worker(), &efd[0]));
+    ASSERT_UCS_OK(ucp_worker_get_efd(p_entity[1]->worker(), &efd[1]));
+
+    polled[0].fd = efd[0];
+    polled[1].fd = efd[1];
+
+    arm(p_entity[0]->worker());
+    arm(p_entity[1]->worker());
+
+    uint64_t recv_data = 0;
+    size_t req_cntr = 0;
+    comp_cntr = 0;
+
+    /* Do not care about data in this test, just count events */
+    for (size_t i = 0; i < msg_count; ++i) {
+        req[i] = ucp_tag_send_nb(p_entity[0]->ep(), &send_data, sizeof(send_data),
+                                 DATATYPE, TAG, send_completion);
+        req_cntr += UCS_PTR_IS_PTR(req[i]) ? 1 : 0;
+    }
+
+    for (size_t i = msg_count; i < 2*msg_count; ++i) {
+        req[i] = ucp_tag_recv_nb(p_entity[1]->worker(), &recv_data, sizeof(recv_data),
+                                 DATATYPE, TAG, (ucp_tag_t)-1, recv_completion);
+        req_cntr += UCS_PTR_IS_PTR(req[i]) ? 1 : 0;
+    }
+
+    /* wait until all messages are completed */
+    for (size_t i = 0; i < 2*msg_count; ++i) {
+        if (UCS_PTR_IS_PTR(req[i])) {
+            wait(req[i]);
+        } else {
+            ASSERT_UCS_OK(UCS_PTR_STATUS(req[i]));
+        }
+    }
+
+    /*
+     * NOTE: ud and udx tests are failing under valgrind w/o
+     *       counting of completions
+     */
+    EXPECT_EQ(req_cntr, comp_cntr);
+
+    int nfd_exp = 0;
+    if (events & (UCP_WAKEUP_TAG_SEND |
+                  UCP_WAKEUP_TAG_RECV /* FIXME: PR #1277 (RNDV) */)) {
```

Comment:
This test is not supposed to send RNDV. Or is something else meant by this FIXME comment?

Reply:
this whole test is a code merge artifact (it was removed from master). will remove.